tests/integration/cmd_link_unlink_test.go

```diff
@@ -9,138 +13,199 @@ import (
 
-				// we link
-				helper.Cmd("odo", "link", "backend", "--context", frontendContext, "--port", "8778").ShouldPass()
-				// ensure that the proper envFrom entry was created
-				envFromOutput := oc.GetEnvFromEntry("frontend", "app", commonVar.Project)
-				Expect(envFromOutput).To(ContainSubstring("backend"))
-
-				dcName := oc.GetDcName("frontend", commonVar.Project)
-				// wait for DeploymentConfig rollout to finish, so we can check if application is successfully running
-				oc.WaitForDCRollout(dcName, commonVar.Project, 20*time.Second)
-				helper.HttpWaitFor(frontendURL, "Hello world from node.js!", 20, 1)
-
-				outputErr := helper.Cmd("odo", "link", "backend", "--port", "8778", "--context", frontendContext).ShouldFail().Err()
-				Expect(outputErr).To(ContainSubstring("been linked"))
-				helper.Cmd("odo", "unlink", "backend", "--port", "8778", "--context", frontendContext).ShouldPass()
+			It("should find the link in odo describe", func() {
+				checkDescribe(frontendContext, backendComp, false, false)
 			})
 
-			It("Wait till frontend dc rollout properly after linking the frontend application to the backend", func() {
-				appName := helper.RandString(7)
-				helper.CopyExample(filepath.Join("source", "nodejs"), frontendContext)
-				helper.Cmd("odo", "create", "nodejs", "--s2i", "frontend", "--app", appName, "--context", frontendContext, "--project", commonVar.Project).ShouldPass()
-				helper.Cmd("odo", "url", "create", "--port", "8080", "--context", frontendContext).ShouldPass()
-				helper.Cmd("odo", "push", "--context", frontendContext).ShouldPass()
-				frontendURL := helper.DetermineRouteURL(frontendContext)
-
-				oc.ImportJavaIS(commonVar.Project)
-				helper.CopyExample(filepath.Join("source", "openjdk"), backendContext)
-				helper.Cmd("odo", "create", "java:8", "--s2i", "backend", "--app", appName, "--project", commonVar.Project, "--context", backendContext).ShouldPass()
-				helper.Cmd("odo", "url", "create", "--port", "8080", "--context", backendContext).ShouldPass()
-				helper.Cmd("odo", "push", "--context", backendContext).ShouldPass()
-
-				// link both component and wait till frontend dc rollout properly
-				helper.Cmd("odo", "link", "backend", "--port", "8080", "--wait", "--context", frontendContext).ShouldPass()
-				helper.HttpWaitFor(frontendURL, "Hello world from node.js!", 20, 1)
-
-				// ensure that the proper envFrom entry was created
-				envFromOutput := oc.GetEnvFromEntry("frontend", appName, commonVar.Project)
-				Expect(envFromOutput).To(ContainSubstring("backend"))
-
-				helper.Cmd("odo", "unlink", "backend", "--port", "8080", "--context", frontendContext).ShouldPass()
+			When("the link is pushed", func() {
+				JustBeforeEach(func() {
+					helper.Cmd("odo", "push", "--context", frontendContext).ShouldPass()
+				})
+				It("should ensure that the proper envFrom entry was created", func() {
+					envFromOutput := oc.GetEnvFromEntry(frontendComp, "app", commonVar.Project, "deployment")
```

Comment:
Do you think we will ever see a DeploymentConfig use `envFrom`? I don't think there is any other resource that would satisfy the `.spec.template.spec.containers[0].envFrom` jsonpath except Deployments and if there is none, I would like to remove the `resourceName` arg and use `Deployment` by default. Also, since we're moving away from --git and --binary, we probably no longer need to interact with DC even if it uses envFrom entry.

I guess,  I will go ahead and remove the `resourceName` arg.

Reply:
My bad, this method was specifically written for DC.